src/store/models/network.ts

```diff
@@ -943,6 +952,60 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  renameNode: thunk(
+    async (actions, { node, newName }, { getState, injections, getStoreActions }) => {
+      if (!newName) throw new Error(l('renameErr', { newName }));
+
+      if (node.status === Status.Started) {
+        await actions.stop(node.networkId);
+      }
+
+      const networks = getState().networks;
+      const network = networks.find(n => n.id === node.networkId);
+      if (!network) throw new Error(l('networkByIdErr', { networkId: node.networkId }));
+
+      let updatedNode;
+      switch (node.type) {
+        case 'lightning':
+          updatedNode = network?.nodes.lightning.find(n => n.id === node.id) as LndNode;
+          getStoreActions().designer.renameNode({
+            name: newName,
+            nodeId: updatedNode.name,
+          });
+          updatedNode.name = newName;
+          updatedNode.paths = getLndFilePaths(newName, network);
+          break;
+        case 'bitcoin':
+          updatedNode = network?.nodes.bitcoin.find(n => n.id === node.id) as BitcoinNode;
+          getStoreActions().designer.renameNode({
+            name: newName,
+            nodeId: updatedNode.name,
+          });
+          updatedNode.name = newName;
+          break;
+        case 'tap':
+          updatedNode = network?.nodes.tap.find(n => n.id === node.id) as TapdNode;
+          getStoreActions().designer.renameNode({
+            name: newName,
+            nodeId: updatedNode.name,
+          });
+          updatedNode.name = newName;
+          updatedNode.paths = getTapdFilePaths(newName, network);
+          break;
+        default:
+          throw new Error('Invalid node type');
+      }
+
+      await injections.dockerService.updateDirs(network, node, newName);
+
+      actions.setNetworks([...networks]);
+      await actions.save();
+
+      await injections.dockerService.saveComposeFile(network);
+
+      await actions.start(node.networkId);
```

Comment:
If the network is stopped and I rename a node, the network starts up afterwards. It should remain in the state it was originally in.